src/components/NewTodo.tsx

```diff
@@ -0,0 +1,83 @@
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+} from 'react';
+import { postTodo } from '../api/todos';
+import { AuthContext } from './Auth/AuthContext';
+import { TitleContext } from './Context/TitleContext';
+
+interface Props {
+  loadTodos: () => Promise<void>,
+  handleError: (errorText: string) => void,
+  isAdding: boolean,
+  setIsAdding: (adding: boolean) => void,
+}
+
+export const NewTodo: React.FC<Props> = React.memo(({
+  loadTodos,
+  handleError,
+  isAdding,
+  setIsAdding,
+}) => {
+  const user = useContext(AuthContext);
+  const newTodoField = useRef<HTMLInputElement>(null);
+  const { todoTitle, setTodoTitle } = useContext(TitleContext);
+
+  useEffect(() => {
+    if (newTodoField.current) {
+      newTodoField.current.focus();
+    }
+  }, []);
+
+  const addTodo = useCallback(async (title: string) => {
+    if (title.trim().length === 0) {
+      handleError('Title can\'t be empty');
+    }
+
+    if (user && title.trim().length > 0) {
+      setIsAdding(true);
+
+      try {
+        await postTodo({
+          userId: user.id,
+          title,
+          completed: false,
+        });
```

Comment:
better use `title.trim()` in this case, to trim redundant spaces. 8 and 9 point of checkList)))

8. NewTodo form shouldn’t create empty todos.
9. NewTodo form should trim redundant spaces.
```suggestion
      try {
        await postTodo({
          userId: user.id,
          title: title.trim(),
          completed: false,
        });
```